src/main/java/graphql/nadel/engine/ServiceResultNodesToOverallResult.java

```diff
@@ -324,6 +324,11 @@ private HandleResult unapplyTransformations(ExecutionId executionId,
                                                 Set<ResultPath> hydrationInputPaths) {
 
         if (isArtificialHydrationNode(node.getFieldIds(), new HashSet<String>(transformationToFieldId.values()))) {
+            if (getFieldIdsWithoutTransformationId(node, transformationMetadata).size() != 0) {
+                HandleResult handleResult = HandleResult.simple(nodesWithTransformationIds(node, null, transformationMetadata));
+                handleResult.traversalControl = TraversalControl.ABORT;
+                return handleResult;
+            }
```

Comment:
fixes a bug when an artificial field is also declared as a not-transformed field